### ISSUE TITLE
[build/include_order] Spurious warning

### ISSUE THREAD

**Issue by alejandro-colomar** [cpplint/cpplint#209]
I'm building the example programs in the EXAMPLES sections in the Linux man-pages, to test their correctness.

As part of that, I use cpplint(1).  In some pages, I've seen the following warnings, which I can't understand how they are being triggered.

`$ cat tmp/src/man3/cacos.3.d/cacos.c`
```c

/* Link with "-lm" */

#include <complex.h>
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>

int
main(int argc, char *argv[])
{
    double complex z, c, f;
    double complex i = I;

    if (argc != 3) {
        fprintf(stderr, "Usage: %s <real> <imag>\n", argv[0]);
        exit(EXIT_FAILURE);
    }

    z = atof(argv[1]) + atof(argv[2]) * I;

    c = cacos(z);

    printf("cacos() = %6.3f %6.3f*i\n", creal(c), cimag(c));

    f = -i * clog(z + i * csqrt(1 - z * z));

    printf("formula = %6.3f %6.3f*i\n", creal(f), cimag(f));

    exit(EXIT_SUCCESS);
}

```
```sh
alx@asus5775:~/src/linux/man-pages/man-pages$ make tmp/src/man3/cacos.3.d/cacos.lint-c.cpplint.touch V=1
LINT (cpplint)	tmp/src/man3/cacos.3.d/cacos.lint-c.cpplint.touch
cpplint   tmp/src/man3/cacos.3.d/cacos.c >/dev/null
tmp/src/man3/cacos.3.d/cacos.c:5:  Found C system header after C++ system header. Should be: cacos.h, c system, c++ system, other.  [build/include_order] [4]
tmp/src/man3/cacos.3.d/cacos.c:6:  Found C system header after C++ system header. Should be: cacos.h, c system, c++ system, other.  [build/include_order] [4]
tmp/src/man3/cacos.3.d/cacos.c:7:  Found C system header after C++ system header. Should be: cacos.h, c system, c++ system, other.  [build/include_order] [4]
make: *** [lib/lint-c.mk:64: tmp/src/man3/cacos.3.d/cacos.lint-c.cpplint.touch] Error 1
```

There are no C++ headers at all.

**Comment by alejandro-colomar** [comment]
Maybe this is the problematic line?:

```c
fprintf(stderr, "Usage: %s <real> <imag>\n", argv[0]);
```

**Comment by tkruse** [comment]
The issue seems to be that `complex.h` is recognized as a C++ header. `cpplint.py` lists it here and here:
* https://github.com/cpplint/cpplint/blob/develop/cpplint.py#L411
* https://github.com/cpplint/cpplint/blob/develop/cpplint.py#L556

I am not sure what's the best way to resolve this.

**Comment by alejandro-colomar** [comment]
> The issue seems to be that `complex.h` is recognized as a C++ header. `cpplint.py` lists it here and here:
> 
>     * https://github.com/cpplint/cpplint/blob/develop/cpplint.py#L411
> 
>     * https://github.com/cpplint/cpplint/blob/develop/cpplint.py#L556
> 
> 
> I am not sure what's the best way to resolve this.

Since it's listed as a legacy C++ header, I'd just remove it.  It's incompatible with C's one, which is not legacy.  In fact, it should be impossible to have the old C++ header in current systems, due to C++ being compatible with C, at least in that regard.

**Comment by tkruse** [comment]
My worry is that some old projects using cpplint out there are using those headers and would start to get warnings. I cannot tell how many users would be impacted (and how to weigh their troubles with the troubles of users of the C system lib). I guess a better way would be to maybe make checking against the legacy C++ headers optional as opt-in, so users of legacy system can keep the original cpplint behavior.

**Comment by alejandro-colomar** [comment]
`<complex.h>` was added in C99.  C++, since C++11, also added `<complex.h>` with the C meaning.  C++ programs should probably not rely on the legacy `<complex.h>` with the meaning of `<complex>`, since it will rarely exist.

**Comment by tkruse** [comment]
fair enough, I guess we can remove it from the legacy list without too much pain
